packages/job-worker/src/workers/parent-base.ts

```diff
@@ -187,8 +185,16 @@ export abstract class WorkerParentBase {
 									continue
 								case ThreadStatus.PendingInit:
 									logger.debug(`Re-initialising worker thread: "${this.#queueName}"`)
-									// Reinit the worker
-									await this.initWorker(mongoUri, this.#mongoDbName)
+									// Reinitialize the worker
+									try {
+										await this.initWorker(mongoUri, this.#mongoDbName)
+									} catch (error) {
+										logger.error(`Worker ${this.#prettyName} could not be initialized:`, {
+											data: error,
+										})
+										this.#threadStatus = ThreadStatus.Closed
```

Comment:
This is my only objection here: marking the thread as `Closed` means that there is no way to recover from this state other than restarting Core. I would suggest that, instead of marking it as Closed, add a sleep here, let this loop again and hope that whatever issue caused initWorker to throw (ultimately: what could be the issue? platform hickup? Database down?) to have been resolved outside of the system and retry initializing the thread. Then the system at least tries to heal itself.

Reply:
I agree that making the system able to try to recover is a better approach.
What we saw, was that when we got into the "infinite loop" in this clause, we generated 250.000+ logs an hour from this alone. Adding a _sleep_ will help reduce the amount of logs, and might give the system (and peripherals) time to recover for the next try.

Reply:
@nytamin and I have spent some time looking into alternate solutions here that allow for recovery, and we found a few other edge cases that we have also made fixes for. We made sure to put in some `await sleep(...)` calls at key points so that it does a slow restart/retry loop which stands a chance of recovering without spamming the log too much.